iceoryx_posh/include/iceoryx_posh/popo/smart_chunk.hpp

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_POPO_SMARTCHUNK_HPP
+#define IOX_POSH_POPO_SMARTCHUNK_HPP
+
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_posh/internal/popo/ports/client_server_port_types.hpp"
+#include "iceoryx_hoofs/cxx/type_traits.hpp"
+#include "iceoryx_hoofs/cxx/unique_ptr.hpp"
+
+namespace iox
+{
+namespace popo
+{
+template <typename T, typename H>
+class SubscriberInterface;
+
+
+namespace internal
+{
+/// @brief helper struct for smartchunk
+template <typename TransmissionInterface, typename T, typename H>
+struct SmartChunkPrivateData
+{
+    SmartChunkPrivateData(cxx::unique_ptr<T>&& smartchunkUniquePtr, TransmissionInterface& publisher) noexcept;
+
+    SmartChunkPrivateData(SmartChunkPrivateData&& rhs) noexcept = default;
+    SmartChunkPrivateData& operator=(SmartChunkPrivateData&& rhs) noexcept = default;
+
+    SmartChunkPrivateData(const SmartChunkPrivateData&) = delete;
+    SmartChunkPrivateData& operator=(const SmartChunkPrivateData&) = delete;
+
+    cxx::unique_ptr<T> smartchunkUniquePtr;
+    std::reference_wrapper<TransmissionInterface> publisherRef;
+};
+
+/// @brief specialization of helper struct for smartchunk for const T
+template <typename TransmissionInterface, typename T, typename H>
+struct SmartChunkPrivateData<TransmissionInterface, const T, H>
+{
+    SmartChunkPrivateData(cxx::unique_ptr<const T>&& smartchunkUniquePtr) noexcept;
+
+    SmartChunkPrivateData(SmartChunkPrivateData&& rhs) noexcept = default;
+    SmartChunkPrivateData& operator=(SmartChunkPrivateData&& rhs) noexcept = default;
+
+    SmartChunkPrivateData(const SmartChunkPrivateData&) = delete;
+    SmartChunkPrivateData& operator=(const SmartChunkPrivateData&) = delete;
+
+    cxx::unique_ptr<const T> smartchunkUniquePtr;
+};
+} // namespace internal
+
+///
+/// @brief The SmartChunk class is a mutable abstraction over types which are written to loaned shared memory.
+/// These smartchunks are publishable to the iceoryx system.
```

Comment:
Similar here with `publishable`. Wouldn't something like `transmittable`/`transferable` be more generic and also fit to request-response?

Reply:
Done